pkg/models/model_plan_relation.go

```diff
@@ -0,0 +1,28 @@
+package models
+
+import "github.com/google/uuid"
+
+//IModelPlanRelation is an interface that represents models that are related to a model plan.
+type IModelPlanRelation interface {
+	GetModelPlanID() uuid.UUID
+}
+
+//ModelPlanRelation is a struct meant to be embedded to show that the object should have model plan relations enforced
+type ModelPlanRelation struct {
+	ModelPlanID uuid.UUID `json:"modelPlanID" db:"model_plan_id"`
+}
+
+//GetModelPlanID returns the modelPlanID of the task list section
+func (m ModelPlanRelation) GetModelPlanID() uuid.UUID {
+	return m.ModelPlanID
+}
+
+//CheckAccess fulfills the check accessinterface
+func (m ModelPlanRelation) CheckAccess() {
+
+}
+
+//ICheckAccess is an interface meant to demonstrate that this model should have an access check
+type ICheckAccess interface {
+	CheckAccess()
+}
```

Comment:
I can't seem to find an implementation of `CheckAccess()` on any of our models. Is the presence of this interface acting as a "flag" to set on models that should have access control checks implemented for them?

Reply:
That's correct, I'm essentially using it like a flag. I'm not strongly attached to the implementation, it's definitely not the main use for interfaces. 

Another possible implementation for the same effect would be to add the CheckAccess() method to IBaseStruct which would return a bool, and we could override the function on classes that need the access check, eg BaseTaskListSection etc

Reply:
Yeah - that would mean the function would probably need to take a `store`, but conceptually that might be a bit cleaner, because then you would just be able to call `CheckAccess` in the access control file, rather than calling the store method directly after doing type assertion

Reply:
Hmm, I wasn't thinking about it like that. I was just meaning, you'd implement the function to say that the struct requires an access check. So that would mean instead of seeing if ICheckAccess is implemented, we check true or false from the CheckAccess method.